src/main/java/com/iexec/worker/PingService.java

```diff
@@ -77,15 +78,22 @@ void triggerSchedulerPing() {
      */
     void pingScheduler() {
         log.debug("Sending ping to scheduler");
-        ResponseEntity<String> response = coreClient.ping(loginService.getToken());
-        if (response.getStatusCode() == HttpStatus.UNAUTHORIZED) {
-            loginService.login();
+        String sessionId = null;
+        try {
+            sessionId = coreClient.ping(loginService.getToken());
+        } catch (FeignException e) {
```

Comment:
```suggestion
        } catch (FeignException.Unauthorized e) {
        } catch (FeignException e) {
        }
```

Reply:
@mcornaton To understand, what is wrong with this approach?

Reply:
```
} catch (FeignException e) {
  // if instance of FeignException.Unauthorized
  // else
}
```